esx_service/vmdk_ops.py

```diff
@@ -129,6 +130,9 @@
 # For managing resource locks.
 lockManager = threadutils.LockManager()
 
+# regex for valid volume name
+VALID_VOLUME_NAME_REG = "[a-zA-Z0-9][a-zA-Z0-9_.-]*"
```

Comment:
We do not care what local diver does, so unless there is a good reason for  the change allowed volnames, please remove this code. 

And, as Prashant has mentioned, the parsing is happening in `parse_vol_name` so any changes should be done there.